tests/crypto_crls_cert_chains/host/CMakeLists.txt

```diff
@@ -3,12 +3,85 @@
 
 add_executable(crypto-extra_host host.cpp)
 
+set(DATA_DIR "../data")
+
 add_custom_command(TARGET crypto-extra_host
-    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/../data ${CMAKE_CURRENT_BINARY_DIR}/../data
-)
+            COMMAND rm -rf ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}
+            COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}
+
+            COMMAND ${CMAKE_COMMAND} -E copy  ${CMAKE_CURRENT_SOURCE_DIR}/${DATA_DIR}/root.cnf ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}/root.cnf
+            COMMAND ${CMAKE_COMMAND} -E copy  ${CMAKE_CURRENT_SOURCE_DIR}/${DATA_DIR}/root2.cnf ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}/root2.cnf
+            COMMAND ${CMAKE_COMMAND} -E copy  ${CMAKE_CURRENT_SOURCE_DIR}/${DATA_DIR}/intermediate.cnf ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}/intermediate.cnf
+            COMMAND ${CMAKE_COMMAND} -E copy  ${CMAKE_CURRENT_SOURCE_DIR}/${DATA_DIR}/intermediate2.cnf ${CMAKE_CURRENT_BINARY_DIR}/${DATA_DIR}/intermediate2.cnf
+
+            # The rest of the custom commands create a set of certificates for validating the OE certificate functions. 
+            # It eliminates the need to check-in hardcoded certificates in the source tree
+            # The certificates and CRLs were generated using the following openssl commands.
+            # Creating Root CA private key and self-signed Root CA cert
+            COMMAND openssl genrsa -out ${DATA_DIR}/RootCA1.key.pem
+            COMMAND openssl req -new -x509 -key ${DATA_DIR}/RootCA1.key.pem -out ${DATA_DIR}/RootCA1.crt.pem -days 3650 -subj "/C=US/ST=Ohio/L=Columbus/O=Acme Company/OU=Acme/CN=Root"
+
+            # We want to generate certificates used in this tests with at least one second apart to make it possible to sort those certificate by time 
+            # "Not Before" of a x509 certificate, which has time resolution up to second 
+            # Creating Intermediate CA certificate, signed by the Intermediate CA
+            COMMAND sleep 1
+            COMMAND openssl genrsa -out ${DATA_DIR}/IntermediateCA1.key.pem
```

Comment:
@andschwa Is it better to use many cmake COMMANDs or is it better to have a bash script and then call it to do stuff?

Reply:
I will keep it as it for now. When it comes to porting this CMake file for Windows, we will revisit this topic again-- whether to wrap them into a script or not.